LOC_Election_Dataset_Walkthrough.ipynb

```diff
@@ -0,0 +1,809 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Exploring the United States Election Datasets\n",
+    "\n",
+    "This notebook demonstrates an intermediate approach to exploring and accessing the United States Election Datasets produced by the Library of Congress.\n",
+    "\n",
+    "As an intermediate approach, this notebook uses functions as reusable blocks of code. These functions could potentially be used outside of this notebook to assist in further exploration."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Imports\n",
+    "Below are the Python packages used in this notebook."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import requests # Used to make HTTP requests; this is how we will scrape the Wayback\n",
+    "from bs4 import BeautifulSoup # Used to process the scraped content \n",
+    "import gzip # Used to decompress the gzipped CDX files\n",
+    "import pandas as pd # Used to transform the data into dataframes for computation\n",
+    "from sklearn.feature_extraction.text import CountVectorizer # Used to create a matrix out of a bag of words\n",
+    "from time import sleep # Used to provide a slight pause between Wayback requests"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Defining Global Variables\n",
+    "Here we’re defining two global variables that will be used in the functions to gather and scrape the content.\n",
+    "The represent the base of the URL structure for two of the important web applications we will use: Cloudfront, which is the front end to where the dataset files are location and Wayback, where the actual web resources are accessed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "CLOUDFRONT_BASE_URL = 'https://d2rxokvmqqcpq7.cloudfront.net/'\n",
```

Comment:
Is this the public URL you want to use-long term or is there a human-meaningful name pointing to it?

Reply:
As far as I'm aware, there isn't a human-meaningful name pointing to it. The cloudfront and the redirects were setup by OCIO/Labs.

Reply:
It's probably fine but you might want to talk with Labs about getting a more durable name under labs.loc.gov in case this ever needs to be rebuilt.